pkg/snapshot/clonetree.go

```diff
@@ -0,0 +1,363 @@
+//
+// Use and distribution licensed under the Apache license version 2.
+//
+// See the COPYING file in the root project directory for full text.
+//
+
+package snapshot
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func CloneTreeInto(scratchDir string) error {
```

Comment:
docstring please for all exported functions.

Reply:
done